src/doc/trpl/concurrency.md

```diff
@@ -0,0 +1,371 @@
+`Mutex<T>` type. Here's the second version of our code. It still doesn't work,
+but for a different reason:
+
+```ignore
+use std::thread::Thread;
+use std::io::timer;
+use std::time::Duration;
+use std::sync::Mutex;
+
+fn main() {
+    let mut data = Mutex::new(vec![1u32, 2, 3]);
+
+    for i in 0 .. 2 {
+        let data = data.lock().unwrap();
+        Thread::spawn(move || {
+            data[i] += 1;
+        });
+    }
+
+    timer::sleep(Duration::milliseconds(50));
+}
+```
+
+Here's the error:
+
+```text
+error: the trait `core::marker::Send` is not implemented for the type `core::marker::NoSend`
+```
+
+You see, [`Mutex`](std/sync/struct.Mutex.html) has a
+[`lock`](http://doc.rust-lang.org/nightly/std/sync/struct.Mutex.html#method.lock)
+method which has this siganture:
+
+```ignore
+fn lock(&self) -> LockResult<MutexGuard<T>>
+```
+
+If we [look at the code for MutexGuard](https://github.com/rust-lang/rust/blob/896cb36ecab3eaeb7f101087e030e43771eca5ca/src/libstd/sync/mutex.rs#L163-L170), we'll see
+this:
+
+```ignore
+__marker: marker::NoSend,
+```
+
+Because our guard is `NoSend`, it's not `Send`. Which means we can't actually
+transfer the guard across task boundaries, which gives us our error.
+
+The [`Arc<T>`](std/sync/struct.Arc.html) type is the answer. An `Arc<T>` allows
+us to reference count some data using atomic operations, which will allow our
+code to work. Here's the working version:
+
+```
+use std::sync::{Arc, Mutex};
+use std::thread::Thread;
+use std::io::timer;
+use std::time::Duration;
+
+fn main() {
+    let data = Arc::new(Mutex::new(0u32));
+
+    for _ in range(0, 10) {
+        let data = data.clone();
+        Thread::spawn(move || {
+            let mut data = data.lock().unwrap();
+            *data += 1;
+        });
+    }
+
+    timer::sleep(Duration::milliseconds(50));
+}
+```
+
+We now call `clone()` on our `Arc`, which increases the reference count. This
+reference is then moved into the new thread. Let's examine the body of the
+thread more closely:
+
+```
+# use std::sync::{Arc, Mutex};
+# use std::thread::Thread;
+# use std::io::timer;
+# use std::time::Duration;
+# fn main() {
+#     let data = Arc::new(Mutex::new(0u32));
+# 
+#     for _ in range(0, 10) {
+#         let data = data.clone();
+Thread::spawn(move || {
+    let mut data = data.lock().unwrap();
+    *data += 1;
+});
+#     }
+# }
+```
+
+First, we call `lock()`, which aquires the mutex's lock. Because this may fail,
+it returns an `Option<T>`, and because this is just an example, we `unwrap()`
+it to get a reference to the data. Real code would have more robust error handling
+here. We're then free to mutate it, since we have the lock.
+
+This timer bit is a bit award, however. We have picked a reasonable amount of
+time to wait, but it's entirely possible that we've picked too high, and that
+we could be taking less time. It's also possible that we've picked too low,
+and that we aren't actually finishing this computation.
```

Comment:
Perhaps this should refer back to `Thread::scoped` (which was introduced at the start) as well, because that’s the most obvious way of avoiding the timer in this case.